src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureProvider.php

```diff
@@ -198,7 +200,18 @@ public function getMetadata(Metadata $documentMetadata, StructureMetadata $struc
         }
 
         if ($class->isSubclassOf(ResourceSegmentBehavior::class)) {
-            $indexMeta->setUrlField($this->factory->createMetadataField('resourceSegment'));
+            $field = $this->factory->createMetadataField('resourceSegment');
+            if ($class->isSubclassOf(RedirectTypeBehavior::class)) {
+                $field = new Expression(
+                    sprintf(
+                        '(object.getRedirectType() === %s ? (object.getRedirectTarget() ? object.getRedirectTarget().getResourceSegment() : "") : (object.getRedirectType() === %s ? object.getRedirectExternal() : object.getResourceSegment()))',
```

Comment:
i dont think that we can split the expression it. We have to make the whole expression for all the documents.

i also dont fallback to `object.getResourceSegment()` if the external link is not set. i did i miss something?